slither/solc_parsing/slither_compilation_unit_solc.py

```diff
@@ -742,6 +742,14 @@ def _convert_to_slithir(self) -> None:
                     self._underlying_contract_to_parser[contract].log_incorrect_parsing(
                         f"Impossible to generate IR for {contract.name}.{func.name} ({func.source_mapping}):\n {e}"
                     )
+                except AssertionError as e:
+                    # Provide contex to user about code that failed beyond AssertionError and stack trace.
+                    func_expressions = "\n".join([f'\t{ex}' for ex in func.expressions])
+                    logger.error(
+                        f"Failed to parse {contract.name}.{func.name} ({func.source_mapping}):\n "
+                        f"{func_expressions}")
+                    raise e
```

Comment:
This looks really helpful and is a great idea! Would you mind making a PR with just these changes?

Reply:
done #1873